rskj-core/src/main/java/org/ethereum/core/TransactionExecutor.java

```diff
@@ -516,9 +525,12 @@ private void finalization() {
                 .logs(notRejectedLogInfos)
                 .result(result.getHReturn());
 
+
         // Accumulate refunds for suicides
         result.addFutureRefund(GasCost.multiply(result.getDeleteAccounts().size(), GasCost.SUICIDE_REFUND));
+        // The actual gas subtracted is equal to half of the future refund
         long gasRefund = Math.min(result.getFutureRefund(), result.getGasUsed() / 2);
```

Comment:
Currently when a refund it's performed, it only updates the `mEndGas` (should be named `gasLeftover`). `gasUsed` keeps being the same value, producing inconsistencies and "transforming" `ProgramResult.gasUsed` into `gasNeeded`.

Reply:
fixed here https://github.com/rsksmart/rskj/pull/1548/commits/fceac70a7afe8c65f4e0707774f482b3d001a1af my concern is that this change can breaks consensus rules, but I'm not completely sure, we need to consider this before merging it. 
My guess is that this shouldn't impact because `result.gasUsed()` is only being used by `TransactionExecutionSummary`, but then `TxExecutionSummary.gasUsed` it's never consumed by anyone